tests/unit/src/main/scala/tests/ResourceOperations.scala

```diff
@@ -0,0 +1,80 @@
+package tests
+
+import java.nio.file.StandardCopyOption
+
+import scala.meta.internal.metals.MetalsEnrichments._
+
+import org.eclipse.lsp4j.CreateFile
+import org.eclipse.lsp4j.DeleteFile
+import org.eclipse.lsp4j.RenameFile
+import org.eclipse.lsp4j.ResourceOperation
+
+/**
+ * Client implementation of how to interpret `ResourceOperation` from LSP, used for testing purposes.
+ */
+object ResourceOperations {
+
+  def applyResourceOperation(resourceOperation: ResourceOperation): Unit = {
+    resourceOperation match {
+      case operation: CreateFile =>
+        createFile(operation)
+      case operation: RenameFile =>
+        renameFile(operation)
+      case operation: DeleteFile =>
+        deleteFile(operation)
+      case _ =>
+    }
+  }
+
+  /**
+   * This method should implement how to interpret the CreateFile operation
+   * there is no guarantee that is compliant with the LSP specs
+   * @param operation CreateFile operation
+   */
+  def createFile(operation: CreateFile): Unit = {
+    val uri = operation.getUri
+    val options = Option(operation.getOptions)
+
+    val overwrite: Boolean =
+      options
+        .map(opt => opt.getOverwrite: Boolean)
```

Comment:
Without it, since the `.getOverwrite` method returns a java.lang.Boolean, the return type of the whole `overwrite` is `Any`. With the `: Boolean` there, there is some sort of implicit conversion that makes the `overwrite` a scala Boolean. Not so sure what's causing the error, but it seems to work.
Happy to change it in something more "readable"!

Reply:
Thx for the explanation!

I think that's ok - I have not idea how to make it better. The only alternative here is to call map as `.map[Boolean](..` but I don't see much difference here